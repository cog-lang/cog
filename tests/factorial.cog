// factiorial.cog

// A simple recursive factorial function

func factorial(n : Int) -> Int
{
	if(n <= 1) return 1;
	return n * factorial(n - 1);
}

func testFactorial(n : Int)
{
	printf("%ld! = %ld\n", n, factorial(n));
}

func main(
	argc : Int,
	argv : Ptr<Ptr<Char> >) -> Int
{
	testFactorial(1);
	testFactorial(2);
	testFactorial(3);
	testFactorial(4);
	testFactorial(5);

	return 0;
}