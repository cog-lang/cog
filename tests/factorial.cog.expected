result: 0
stdout: {
1! = 1
2! = 2
3! = 6
4! = 24
5! = 120
}
stderr: {
}
